sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleAssignmentProperties.java

```diff
@@ -23,18 +21,13 @@
      * @param roleDefinitionId The {@link KeyVaultRoleDefinition role definition} ID used in the
      * {@link KeyVaultRoleAssignment role assignment}.
      * @param principalId The principal ID assigned to the role. This maps to the ID inside the Active Directory.
-     * It can point to a user, service principal, or security group.
+     * It can point to a user, service principal, or security group.*
+     * @param roleScope The {@link KeyVaultRoleScope scope} of this {@link KeyVaultRoleAssignment role assignment}.
      */
-    public KeyVaultRoleAssignmentProperties(String roleDefinitionId, String principalId) {
-        Objects.requireNonNull(roleDefinitionId,
-            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
-                "'roleDefinitionId' in 'properties'"));
-        Objects.requireNonNull(principalId,
-            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
-                "'principalId' in 'properties'"));
-
+    public KeyVaultRoleAssignmentProperties(String roleDefinitionId, String principalId, KeyVaultRoleScope roleScope) {
```

Comment:
Why publicly creatable (same comment as above)? And if intended, the param in context could/should probably just be `scope` as well. The `role` prefix was meant for places where it needed to be disambiguated.

Reply:
All these return types in Java have to be public so that deserialization can work properly. Even if users can create them they are not accepted as input parameters anywhere.

Reply:
Will rename to `scope` though.

Reply:
Now that I think of it, given the way we create these types, the constructors can be package private.

Reply:
Actually, in this case it's because the model is in a different package than the client, which means we need to make the constructor and the class `public`.